core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/ldap/ReadOnlyLDAPUserStoreManager.java

```diff
@@ -2639,6 +2650,462 @@ public int getTenantId() throws UserStoreException {
         return values.toArray(new String[values.size()]);
     }
 
+    /**
+     * This method support multi-attribute search filters for user(s).
+     *
+     * @param condition   Validated Condiction tree
+     * @param profileName Default profile name
+     * @param limit       The number of entries to return in a page.
+     * @param offset      Start index.
+     * @param sortBy      Sort according to the given attribute name.
+     * @param sortOrder   Sorting order.
+     * @return A non-null PaginatedSearchResult instance. Typically contains user names with pagination.
+     * @throws UserStoreException If an UserStoreException is encountered
+     *                            while searching for users in a given condition.
+     */
+    protected PaginatedSearchResult doGetUserList(Condition condition, String profileName, int limit, int offset,
```

Comment:
This doGetUserList() method is very long, please break this into several but meaningful methods, ideally each method should have only 20/30 LOC.

Reply:
Fixed.